src/lib/stores.ts

```diff
@@ -23,17 +24,49 @@ export const user = writable<User | undefined>(undefined);
 export const canisters = writable<Canisters | undefined>(undefined);
 export const waterNeuronInfo = writable<WaterNeuronInfo | undefined>(undefined);
 
+/* === Input Amount ==== */
+function createInputAmountStore() {
+	const { subscribe, set } = writable<string>();
+
+	return {
+		subscribe,
+		change: (value: number) => {
+			const input = value.toString().replace(',', '.');
+			set(input);
+		},
+		set: (value: string) => set(value),
+		reset: () => set('')
+	};
+}
+
+export const inputAmount = createInputAmountStore();
+
+export function handleInputAmount(event: Event): void {
+	const target = event.target as HTMLInputElement;
+	const value = target.value;
+	const regex = /^[0-9]*([\.][0-9]*)?$/;
+
+	if (regex.test(value)) {
+		inputAmount.set(value);
+	} else {
+		inputAmount.set(value.substring(0, value.length - 1));
+		target.value = value.substring(0, value.length - 1);
+	}
+}
+
 /* === SNS === */
 function createBoomerangSnsStore() {
 	const { subscribe, set, update } = writable<{
 		name: string;
 		principal: string;
 		encodedAccount: string | undefined;
+		hex: string;
```

Comment:
what's hex?

Reply:
The account identifier hex

Reply:
Then call it `accountIdentifier`?

Reply:
I didn't because I defined the account variable which is the Boomerang account+subaccount and I thought it was misleading +  it's a hex representation

Reply:
However I changed the code to get rid of the confusion